checkov/terraform/graph_builder/graph_components/blocks.py

```diff
@@ -81,6 +81,14 @@ def __init__(
             self.for_each_index: Optional[Any] = None
             self.foreach_attrs: list[str] | None = None
 
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, TerraformBlock):
+            return False
+
+        return self.name == other.name and self.config == other.config and self.path == other.path and \
```

Comment:
let's add a comment in the init func that if adding var to self you need to update this as well